### PR TITLE
Rework DuplicateSelection to fix bugs

### PR DESCRIPTION
**Pull request by cfognom**
Finally managed to fix the bugs with duplicate lines up. Such a feature seems so simple but it was unimaginably hard to do with the VisualStudio API, especially manipulating the selections. But I think I managed to do it and all bugs should be gone.

I also replaced the old references with NuGet package references and updated some packages. Note that there may still be some packages that are not in use (I couldn't find an easy way to check for unused packages).

### CODE REVIEW

**Comment by justcla** [comment]
Dude, you are a legend. This work is exceptional. Gold-standard coding.

Yes, indeed, I struggled all last week with the fact that adding text to the end of the selection would expand the selection. I tried a hack where I add the text to one character after the end point and add the skipped char to the end of the text, then delete the skipped char. It actually worked fairly well, but when the UNDO operation was applied, the original selection would be increased by one char. So frustrating!

But your solution handles that nicely - by manually re-creating the selection, which is something I was trying desperately not to have to do. (Mostly because I didn't know how. Thanks for discovering the IMultiSelectionBroker, and working out how to setSelections.

On a separate note, I love how you provided meaning comments in the implementation.
I like how you cleanly separated those helper functions (and marked one with TODO to extract).

And THANK YOU for fixing all those references.

In summary, I am overwhelmed with how good this PR is. You are my new hero. I want to buy you a beer. 😊

I will merge your PR, update the release notes to give you credit, and push the new version to the Marketplace.

Thank you. Thank you!

**Comment by cfognom** [comment]
Thanks for the kind words! I'm glad you like it.

> I tried a hack where I add the text to one character after the end point and add the skipped char to the end of the text, then delete the skipped char. It actually worked fairly well, but when the UNDO operation was applied, the original selection would be increased by one char. So frustrating!

Exactly this happened to me too, haha! So annoying that for a very rare edge case you have to make a monolithic solution that  rebuilds all selections from scratch.

**Comment by cfognom** [comment]
Hmm, I seem to be getting this popup when I launch VS with the latest build of HotCommands: 
image
Maybe I messed something up with the references?

**Comment by justcla** [comment]
My bad. I get that warning on my machine. I thought it was due to a different extension. I will look at the error and try to fix it.
It might be the Keyboard shortcuts extension, which I embed in the extension.
